x/committee/simulation/operations/msg.go

```diff
@@ -0,0 +1,185 @@
+		// num voters determined by whether the proposal should pass or not
+		numMembers := int64(len(randomCommittee.Members))
+		majority := randomCommittee.VoteThreshold.Mul(sdk.NewInt(numMembers).ToDec()).Ceil().TruncateInt64()
+		var numVoters int64 = 0
+		// Catch the edge case where vote threshold is zero.
+		// This can be interpreted as meaning a vote should pass immediately after submition even with no votes.
+		// There is no numVoters that will make the proposal not pass, so just leave it at zero and let the proposal pass.
+		if majority > 0 {
+			numVoters = r.Int63n(majority) // in interval [0, majority)
+		}
+
+		shouldPass := r.Float64() < proposalPassPercentage
+		if shouldPass {
+			numVoters = majority + r.Int63n(numMembers-majority+1) // in interval [majority, numMembers]
+		}
+		voters := randomCommittee.Members[:numVoters]
+
+		// schedule vote operations
+		var futureOps []simulation.FutureOperation
+		for _, v := range voters {
+			rt, err := RandomTime(r, ctx.BlockTime(), proposal.Deadline)
+			if err != nil {
+				return noOpMsg, nil, fmt.Errorf("random time generation failed: %w", err)
+			}
+			fop := simulation.FutureOperation{
+				BlockTime: rt,
+				Op:        SimulateMsgVote(keeper, v, proposal.ID),
+			}
+			futureOps = append(futureOps, fop)
+		}
+
+		return submitOpMsg, futureOps, nil
+	}
+}
+
+func SimulateMsgVote(keeper committee.Keeper, voter sdk.AccAddress, proposalID uint64) simulation.Operation {
+	handler := committee.NewHandler(keeper)
+
+	return func(r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, accs []simulation.Account) (
+		opMsg simulation.OperationMsg, fOps []simulation.FutureOperation, err error) {
+
+		msg := committee.NewMsgVote(voter, proposalID)
+		if msg.ValidateBasic() != nil {
+			return noOpMsg, nil, fmt.Errorf("expected msg to pass ValidateBasic: %s", msg.GetSignBytes())
+		}
+
+		res := submitMsg(ctx, handler, msg)
+		return simulation.NewOperationMsg(msg, res.IsOK(), res.Log), nil, nil
+	}
+}
+
+func submitMsg(ctx sdk.Context, handler sdk.Handler, msg sdk.Msg) sdk.Result {
+	ctx, write := ctx.CacheContext()
+	result := handler(ctx, msg)
+	if result.IsOK() {
+		write()
+	}
+	return result
+}
+
+// TODO move random funcs to common location
+
+func RandomTime(r *rand.Rand, inclusiveMin, exclusiveMax time.Time) (time.Time, error) {
+	if exclusiveMax.Before(inclusiveMin) {
+		return time.Time{}, fmt.Errorf("max must be > min")
+	}
+	period := exclusiveMax.Sub(inclusiveMin)
+	subPeriod, err := RandomPositiveDuration(r, 0, period)
+	if err != nil {
+		return time.Time{}, err
+	}
+	return inclusiveMin.Add(subPeriod), nil
+}
+
+func RandomPositiveDuration(r *rand.Rand, inclusiveMin, exclusiveMax time.Duration) (time.Duration, error) {
+	min := int64(inclusiveMin)
+	max := int64(exclusiveMax)
+	if min < 0 || max < 0 {
+		return 0, fmt.Errorf("min and max must be positive")
+	}
+	if min >= max {
+		return 0, fmt.Errorf("max must be > min")
+	}
+	randPositiveInt64 := r.Int63n(max-min) + min
```

Comment:
I didn't use `simulation.RandIntBetween` because it panicked when calling with min and max =0. I think I hit this case in some places, and wanted it to return 0 (which is still valid for an inclusive range). Then doing that also extended it to work with negative numbers.
Though in v0.38 upgrade it's not used anymore.